test/Feature/UpsertSpec.hs

```diff
@@ -92,6 +92,14 @@ spec =
           `shouldRespondWith`
           [json|[ { "a": "1", "b": "0" } ]|] { matchStatus = 201 , matchHeaders = [matchContentTypeJson] }
 
+      it "succeeds if not a single resource is created" $ do
+        request methodPost "/tiobe_pls" [("Prefer", "return=representation"), ("Prefer", "resolution=ignore-duplicates")]
+          [json|[ { "name": "Java", "rank": 1 } ]|] `shouldRespondWith`
+          [json|[]|] { matchStatus = 201 , matchHeaders = [matchContentTypeJson] }
```

Comment:
If no records got created, then a [201](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.2.2) response doesn't seem accurate. Maybe a 200 in this case?

If it's a pain to change then it maybe it's not such a big deal. What do you think?

Reply:
Actually tried that before by applying a `queryTotal == 0` condition but other unrelated tests failed.

Like this [one](https://github.com/begriffs/postgrest/blob/master/test/Feature/InsertSpec.hs#L137), the issue is the `return=minimal` always makes the `queryTotal == 0` despite rows being inserted.

There's another way of detecting the insertion by using the [xmax](https://stackoverflow.com/a/39204667/4692662) system column, tried to use that for PUT as well (since it should respond 200 for update and 201 for insert according to RFC) but it involves a not-so-easy refactoring.

How about if I merge as it is and open an issue for that to correct it later?

Reply:
Also some old insert tests like this one https://github.com/begriffs/postgrest/blob/master/test/Feature/InsertSpec.hs#L129-L135 also respond with 201 despite not inserting nothing.

Reply:
OK, since it's an existing issue, we can merge this PR and fix that one in its own.

(Cool xmax trick btw!)